client/state/selectors/get-sites.js

```diff
@@ -20,7 +20,7 @@ export default createSelector(
 	( state ) => {
 		const primarySiteId = getPrimarySiteId( state );
 		const [ primarySite, sites ] = partition( state.sites.items, { ID: primarySiteId } );
-		return map( concat( primarySite, sites ), site => getSite( state, site.ID ) );
+		return map( concat( primarySite, sortBy( sites, [ 'name', 'URL' ] ) ), site => getSite( state, site.ID ) );
```

Comment:
The regular implementation of `sortBy` takes the collection as its first argument, so you'd need that in place; plus, `primarySite` is a single element so it wouldn't have the ellipsis preceding it. Otherwise, I'm good with that change. :)